src/validator.cpp

```diff
@@ -291,6 +293,50 @@ struct Validator::ValidatorImpl
     void checkUnitForCycles(const ModelPtr &model, const UnitsPtr &parent,
                             std::vector<std::string> &history,
                             std::vector<std::vector<std::string>> &errorList);
+
+    /** @brief Function to check ID fields set within the model scope for duplicates.
+     *
+     * @param model The model to be checked.
+     */
+    void checkForDuplicateIds(const ModelPtr &model);
```

Comment:
for me, this should be more more like `checkUniqueIds`

Reply:
aa2e8d22